src/main/java/at/hannibal2/skyhanni/features/misc/customscoreboard/CustomScoreboard.kt

```diff
@@ -0,0 +1,114 @@
+//
+// Requested by alpaka8123 (https://discord.com/channels/997079228510117908/1162844830360146080)
+// Done by J10a1n15, with lots of help from hanni and more contribs <3
+// Also big thanks to the people that tested it before it released, saved me some time <3
+//
+
+//
+// TODO LIST
+// V2 RELEASE
+//  - Soulflow API
+//  - Bank API (actually maybe not, I like the current design)
+//  - beacon power
+//  - skyblock level
+//  - more bg options (round, blurr, outline)
+//  - countdown events like fishing festival + fiesta when its not on tablist
+//
+
+package at.hannibal2.skyhanni.features.misc.customscoreboard
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.AlignmentEnum
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAlignedWidth
+import net.minecraftforge.client.GuiIngameForge
+import net.minecraftforge.client.event.RenderGameOverlayEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+private val config get() = SkyHanniMod.feature.gui.customScoreboard
+private var display = emptyList<Pair<String, AlignmentEnum>>()
+
+class CustomScoreboard {
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isCustomScoreboardEnabled()) return
+        if (display.isEmpty()) return
+        val position = config.position
+
+        RenderBackground().renderBackground()
+
+        position.renderStringsAlignedWidth(display, posLabel = "Custom Scoreboard")
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isCustomScoreboardEnabled()) return
+
+        // Creates the lines
+        display = createLines()
```

Comment:
How many startsWith() calls does this one function call cause? per tick? this feels super inperformant if im not missing something. Please use a combination of time interval update and scoreboard update, best done with a `dirty` flag that gets checked every 5ish ticks or so

Reply:
[Like this?](https://github.com/hannibal002/SkyHanni/pull/699/commits/97cec4e341daebfb7d6d298fb1053c4993ffe620)